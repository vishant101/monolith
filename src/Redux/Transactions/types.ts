export interface TransactionState {
  status: string;
  favourites: Array<number>;
  processedTransactions: ProcessedTransactions;
  error: string;
}

export interface Transaction {
  user_id: string;
  timestamp: string;
  currency: string;
  amount: string;
}

export interface Transactions extends Array<Transaction> {}

export interface Balances {
  [currency: string]: string;
}

export interface ProcessedTransactionProps {
  user_id: string;
  lastupdated: string;
  transactions: Transactions;
  balances: Balances;
}

export interface ProcessedTransactions {
  [user_id: string]: ProcessedTransactionProps;
}
