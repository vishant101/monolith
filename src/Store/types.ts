import {TransactionState} from '../Redux/Transactions/types';

export interface RootState {
  transactionState: TransactionState;
}
