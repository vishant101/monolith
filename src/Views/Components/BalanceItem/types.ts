import {Balances} from '../../../Redux/Transactions/types';

export interface Props {
  user_id: string;
  balances: Balances;
  timestamp: string;
  onPress: () => void;
}
