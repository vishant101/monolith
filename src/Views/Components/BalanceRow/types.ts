export interface Props {
  currency: string;
  balance: string;
}
