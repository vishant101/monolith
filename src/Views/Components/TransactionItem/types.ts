export interface Props {
  currency: string;
  amount: string;
  timestamp: string;
}
