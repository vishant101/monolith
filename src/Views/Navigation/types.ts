export type StackParamList = {
  Balances: undefined;
  Transactions: {user_id: string};
};
