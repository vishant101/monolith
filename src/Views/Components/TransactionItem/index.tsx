import React from 'react';
import {Text, Image} from 'react-native';

import {getCurrencyImgSource} from '../../../Utils';
import {
  Container,
  CurrencyContainer,
  AmountText,
  CurrencyText,
  styles,
} from './styles';
import {Props} from './types';

const TransactionItem = ({currency, amount, timestamp}: Props) => {
  return (
    <Container>
      <Text>{timestamp}</Text>
      <CurrencyContainer>
        <AmountText>{amount}</AmountText>
        <CurrencyText>{currency}</CurrencyText>
        <Image
          source={getCurrencyImgSource(currency)}
          style={styles.image}
          resizeMode="contain"
        />
      </CurrencyContainer>
    </Container>
  );
};

export default TransactionItem;
