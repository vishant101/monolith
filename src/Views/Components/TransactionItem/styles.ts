import styled from 'styled-components/native';

export const Container = styled.View`
  justify-content: space-between;
  flex-direction: row;
  padding: 15px 20px;
  background-color: white;
`;

export const CurrencyContainer = styled.View`
  justify-content: space-between;
  flex-direction: row;
`;
