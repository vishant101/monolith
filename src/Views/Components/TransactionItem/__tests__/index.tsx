import React from 'react';
import renderer from 'react-test-renderer';
import {render} from '@testing-library/react-native';

import TransactionItem from '..';
import {Props} from '../types';

describe('TransactionItem Component', () => {
  let props: Props;

  beforeEach(() => {
    props = {
      currency: 'EUR',
      amount: '1000',
      timestamp: '10000',
    };
  });

  it('should render successfully', () => {
    expect(render(<TransactionItem {...props} />).toJSON()).toBeTruthy();
  });

  it('should match the previous snapshot', () => {
    expect(renderer.create(<TransactionItem {...props} />)).toMatchSnapshot();
  });
});
