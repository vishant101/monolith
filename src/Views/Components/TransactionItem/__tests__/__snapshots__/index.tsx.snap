// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionItem Component should match the previous snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingBottom": 15,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 15,
      },
    ]
  }
>
  <Text>
    10000
  </Text>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "paddingBottom": 0,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 0,
          },
        ]
      }
    >
      1000
    </Text>
    <Text
      style={
        Array [
          Object {
            "paddingBottom": 0,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 0,
          },
        ]
      }
    >
      EUR
    </Text>
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../src/Assets/Img/ic_eur.png",
        }
      }
      style={
        Object {
          "height": 24,
          "resizeMode": "stretch",
          "width": 24,
        }
      }
    />
  </View>
</View>
`;
